src/utils/useRoutedTabs.js

```diff
@@ -7,15 +7,28 @@ import { useHistory, useLocation } from 'react-router-dom'
 const useRoutedTabs = (tabPaths, defaultTabIndex = 0) => {
   const { pathname } = useLocation()
   const history = useHistory()
+
+  // get breadcrumbs
+  const segments = pathname.split('/')
+
   const [tabIndex, setTabIndex] = useState(() => {
     // Set the initial tabIndex from the URL on page load
-    const matchedIndex = tabPaths.indexOf(pathname)
+    const matchedIndex = tabPaths.findIndex((tabName) =>
+      pathname.includes(tabName),
+    )
     return matchedIndex === -1 ? defaultTabIndex : matchedIndex
   })
 
   const handleTabChange = (tabIndex) => {
     setTabIndex(tabIndex)
-    history.push(tabPaths[tabIndex])
+
+    // on shallow match return to shallow root
+    if (segments[1] === tabPaths[tabIndex].slice(1)) {
```

Comment:
Same with matchPath here, with exact set to true. The reason is because `/mapasdf` could possibly match with `/map`.